tagcompare/logger.py

```diff
@@ -0,0 +1,56 @@
+import os
+import logging
+
+import settings
+
+
+class Logger(object):
+    def __init__(self, name, directory=None, writefile=False):
+        if directory and not writefile:
+            raise ValueError("directory was specified when writefile is False!")
+
+        self.__directory = directory
+        if writefile:
+            if not directory:
+                self.__directory = os.path.join(
+                    os.path.abspath(os.path.dirname(__file__)), settings.LOG_DIR)
+            if not os.path.exists(self.__directory):
+                raise IOError("directory doesn't exist at {}".format(self.__directory))
+
+        self.__name = name.replace('.log', '')
+        logger = logging.getLogger('tagcompare.%s' % name)
+
+        # Set default log levels
+        logging.getLogger("tagcompare").setLevel(logging.DEBUG)
+        if not logger.handlers:
+            if writefile:
+                logger.addHandler(self.__file_handler())
+            logger.addHandler(self.__stream_handler())
+        self._logger = logger
+
+    @property
+    def name(self):
+        return self.__name
```

Comment:
Looks like this is never used. `test_init()` is getting the name of the thing the `.get()` method returns, not an object of this class.

I'd also suggest renaming this class to something like `LoggerFactory`, since it isn't actually a logger -- it's a thing for making loggers.

Or maybe it doesn't need to be a class at all -- the `logging` module already provides a log object which is the thing a user of this module is interested in. I don't see anything outside of the tests that does anything with a `Logger` object except `Logger(...).get()`, and most of `Logger`'s methods don't even reference `self`. Except the attribute getters, which don't seem to be used externally.